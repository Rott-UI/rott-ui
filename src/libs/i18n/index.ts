export * from './models'
export {default as tr} from './tr-TR.json'
export {default as en} from './en-US.json'
