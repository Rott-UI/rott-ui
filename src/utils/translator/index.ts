export * from './translator'
export * from './models'
