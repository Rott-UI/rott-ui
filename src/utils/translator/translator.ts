/* eslint-disable react-hooks/rules-of-hooks */

// Util and Lib Imports
import '../stringExtensions'
import '../arrayExtensions'
import {TranslationLanguageTypes} from './models'

// Package Imports
import {MessageDescriptor, PrimitiveType, createIntl, createIntlCache, useIntl} from 'react-intl'

// Provider Imports
import {languageMessages, themeConfig} from '../../providers'

/**
 *
 * @param {TranslationLanguageTypes} key Dil dosyasında yazılı olan anahtar kelime. Örn: Common.Hello -> Türkçe Merhaba, İngilizce Hello şeklinde çıktı verir.
 * @param {Record<string, PrimitiveType | any | undefined>} extraMessages Dil dosyasında kullanılan anahtar keline eğer ekstra parametre alıyorsa obje şeklinde verilmelidir. Örn, {name: 'Cindoruk'}}.
 *
 * @returns {string} Seçilen dil içinde yer alan anahtar kelimeye karşılık geren çeviriyi ekrana getirir. Verilen key seçilen dil dosyasında yoksa "failed" olarak sonuç döner.
 *
 */
export const translator = (
  key: TranslationLanguageTypes,
  extraMessages?: Record<string, PrimitiveType | any | undefined>
): string => {
  const intl = useIntl()

  if (!key?.isEmpty()) return intl?.formatMessage({id: `${key}`}, extraMessages)
  else return 'failed'
}

export const formatMessage = (
  message: TranslationLanguageTypes,
  extraMessages?: Record<string, PrimitiveType | any | undefined>,
  descriptor?: MessageDescriptor
): string => {
  const cache = createIntlCache()

  // TODO: bu alan dil paketi eklendikten sonra duzeltilecek
  const selectedLang = themeConfig.options.appLanguage //store.getState().app.lang

  const intl = createIntl(
    {
      locale: selectedLang?.name ?? 'tr-TR',
      messages:
        Object.entries(languageMessages).find((key) => key[0] === selectedLang?.name)?.[1] ??
        languageMessages['tr-TR'],
    },
    cache
  )

  return intl.formatMessage(descriptor ? descriptor : {id: `${message}`}, extraMessages)
}
