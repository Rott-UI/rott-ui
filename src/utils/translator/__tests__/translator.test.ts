// Util and Lib Imports
import {en, tr} from '../../../libs'
import {formatMessage, translator} from '../translator'

describe('Util -> Translator', () => {
  it('english dil snapshotı eşleşmeli', () => {
    expect(en).toMatchSnapshot()
  })

  it('türkçe dil snapshotı eşleşmeli', () => {
    expect(tr).toMatchSnapshot()
  })

  it('translator metoduna verilen geçerli değeri ekranda göstermeli', () => {
    const translatedString = translator('TEST')

    expect(translatedString).toBe(en.TEST)
  })

  it('türkçe dil keyleri english dil keyleri ile eşleşmeli', () => {
    const turkishLanguageKeys = Object.keys(tr)
    const englishLanguageKeys = Object.keys(en)

    expect(turkishLanguageKeys).toStrictEqual(englishLanguageKeys)
  })

  it('translator verilen key ile birlikte ekstra mesaj yazılması gerekiyorsa onu birleştirip dönmelidir.', () => {
    const translatedString = translator('TEST.WITH.PARAM', {testText: 'test'})

    expect(translatedString).toBe('TEST WITH test')
  })
})

describe('Util -> FormatMessage', () => {
  it('formatMessage metoduna verilen geçerli değeri ekranda göstermeli', () => {
    const translatedString = formatMessage('TEST', undefined, undefined)

    expect(translatedString).toBe(en.TEST)
  })
})
