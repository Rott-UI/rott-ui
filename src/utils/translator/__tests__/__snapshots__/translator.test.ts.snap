// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Util -> Translator english dil snapshotı eşleşmeli 1`] = `
{
  "COMMON.ACCOUNT.SELECTION": "Account Selection",
  "COMMON.AUTOMATIC": "Automatic",
  "COMMON.BACK": "Back",
  "COMMON.CANCEL": "Cancel",
  "COMMON.CLEAR": "Clear",
  "COMMON.CONFIRM": "Confirm",
  "COMMON.CONTINUE": "Continue",
  "COMMON.COPY": "Copied",
  "COMMON.DEAR": "Dear",
  "COMMON.EDIT": "Edit",
  "COMMON.ERROR": "Error!",
  "COMMON.INFO": "Information",
  "COMMON.LAST.3.MONTHS": "Last 3 Months",
  "COMMON.LAST.6.MONTHS": "Last 6 Months",
  "COMMON.LOGIN": "Sign In",
  "COMMON.LOGOUT": "Logout",
  "COMMON.OK": "Ok",
  "COMMON.PLEASE.WAIT": "Please Wait",
  "COMMON.PROCESS": "Process",
  "COMMON.RECIPIENT.DETAILS": "Recipient Details",
  "COMMON.SEARCH": "Search",
  "COMMON.SELECT.DATE": "Select Date",
  "COMMON.SUBMIT": "Submit",
  "COMMON.THIS.MONTH": "This Month",
  "COMMON.THIS.WEEK": "This Week",
  "COMMON.THIS.YEAR": "This Year",
  "COMMON.TODAY": "Today",
  "COMMON.TOMORROW": "Tomorrow",
  "COMMON.TRY_AGAIN": "Please try againg later!",
  "COMMON.UNKNOWN": "Unknown",
  "COMMON.WARNING": "Warning",
  "CVC.CODE": "CVC Code",
  "CVC.INFO.DESCRIPTION": "The Card Verification Code or CVC* is an additional code printed on your bank or credit card. It is the last three digits of the number written on the signature strip on the back of your card.",
  "CVC.INFO.TITLE": "CVC - Card Verification Code",
  "DAY.1": "Monday",
  "DAY.2": "Tuesday",
  "DAY.3": "Wednesday",
  "DAY.4": "Thursday",
  "DAY.5": "Friday",
  "EXPIRE.DATE.PLACEHOLDER": "Month / Year",
  "EXPIRE.DATE.TITLE": "Expire Date",
  "INPUT.ACCOUNTNUMBER": "Account Number",
  "INPUT.AMOUNT": "Transfer Amount",
  "INPUT.BANKACCOUNTNUMBER": "Bank Account Number",
  "INPUT.CARD.INFO": "Card Info",
  "INPUT.CREDITCARD": "Credit Card",
  "INPUT.CUSTOMERNUMBER": "Customer Number",
  "INPUT.CVC": "CVC",
  "INPUT.DATE": "Date",
  "INPUT.DATETIME": "DateTime",
  "INPUT.DROPDOWN": "Select",
  "INPUT.DROPDOWN.SEARCH": "{name} Search ...",
  "INPUT.EXPIREDATE": "Expire Date",
  "INPUT.IBAN": "Iban",
  "INPUT.INSTITUTION.NAME": "Institution Name",
  "INPUT.MYACCOUNT": "My Account",
  "INPUT.NAME": "Enter Name",
  "INPUT.NUMERIC": "Numeric",
  "INPUT.PASSWORD": "Password",
  "INPUT.PASSWORD.CHANGE.PERIOD": "Password Change Period",
  "INPUT.PASSWORD.CONFIRM": "Confirm Password",
  "INPUT.PHONE": "Phone",
  "INPUT.PHONE.NUMBER": "Phone Number",
  "INPUT.PINPASSWORD": "Pin Password",
  "INPUT.PLATENUMBER": "Plate Number",
  "INPUT.RECORDNAME": "Enter Quick Transaction Name",
  "INPUT.SELECT.INPUT": "Select {name}",
  "INPUT.STATEMENT": "Enter a Description",
  "INPUT.SUBSCRIBER.NUMBER": "Subscriber Number",
  "INPUT.SURNAME": "Enter Surname",
  "INPUT.TCNUMBER": "TC Identity Number",
  "INPUT.TIME": "Time",
  "INPUT.USERNUMBER": "User Number",
  "LIST.EMPTY.WARN": "No data found!",
  "LIST.ITEM.FAVORITE.ADD": "Add to Favorites",
  "LIST.ITEM.FAVORITE.REMOVE": "Remove from Favorites",
  "LOGOUT.DESCRIPTION": "Are you sure you want to make an exit?",
  "LOGOUT.TITLE": "Log Out",
  "NEW.PASSWORD.SCREEN.SUBMIT": "Save New Password",
  "NEW.PASSWORD.SCREEN.TITLE": "Create New Password",
  "NOTIFICATION.CLEAR.ALL": "Clear All",
  "NOTIFICATION.EMPTY.WARN": "Today or prior, there are no intraday notifications recorded!",
  "NOTIFICATION.TITLE": "Notifications",
  "OTP.DIGIT.REQUIREMENT": "Must be exactly 6 digits",
  "OTP.INSTRUCTIONS": "Please enter the code
sent to your phone number {phoneNumber}.",
  "OTP.INSTRUCTIONS.NO.PHONE": "Enter the verification code sent to your phone",
  "OTP.REMAININGTIME": "Time remaining for verification",
  "OTP.RESENDCODE": "Resend",
  "OTP.SMS.VERIFICATION.CODE": "SMS verification code",
  "OTP.SMSNOTRECEIVED": "Did you not receive the SMS?",
  "OTP.TITLE": "Please enter the SMS verification code sent to your phone",
  "OTP.VERIFY": "Verify",
  "PROFILE.ARENTYOU": "Aren't You?",
  "PROFILE.LOGOUT.DESCRIPTION": "Are you sure you want to switch accounts?",
  "PROFILE.LOGOUT.TITLE": "Confirm Account Change",
  "SELECT.MONTH": "{month} Month",
  "TEST": "TEST",
  "TEST.WITH.PARAM": "TEST WITH {testText}",
  "VALIDATION.CONFIRMATION": "{name} confirmation is required.",
  "VALIDATION.CONFIRMATION_NOT_MATCH": "{name} and {confirmName} didn't match.",
  "VALIDATION.EMAIL_FORMAT": "Wrong email format.",
  "VALIDATION.INVALID": "Invalid {name}.",
  "VALIDATION.INVALID.AMOUNT_NOT_MULTIPLE_OF_10": "You can send amounts in multiples of 10 TL.",
  "VALIDATION.INVALID.FORMAT": "Invalid {name} format.",
  "VALIDATION.IP_ADDRESS": "IP Address is not valid.",
  "VALIDATION.MAX_LENGTH": "Maximum {length} characters.",
  "VALIDATION.MAX_NUMBER": "{name} must be less than or equal to {number}",
  "VALIDATION.MIN_LENGTH": "Minimum {length} characters.",
  "VALIDATION.NUMBER": "{name} must be a number.",
  "VALIDATION.POSITIVE": "{name} has to be a positive number.",
  "VALIDATION.REQUIRED": "{name} is required.",
  "VALIDATION.SELECT": "You must choose.",
}
`;

exports[`Util -> Translator türkçe dil snapshotı eşleşmeli 1`] = `
{
  "COMMON.ACCOUNT.SELECTION": "Account Selection",
  "COMMON.AUTOMATIC": "Otomatik",
  "COMMON.BACK": "Geri",
  "COMMON.CANCEL": "İptal",
  "COMMON.CLEAR": "Temizle",
  "COMMON.CONFIRM": "Uygula",
  "COMMON.CONTINUE": "Devam",
  "COMMON.COPY": "Kopyalandı",
  "COMMON.DEAR": "Sayın",
  "COMMON.EDIT": "Düzenle",
  "COMMON.ERROR": "Hata!",
  "COMMON.INFO": "Bilgilendirme",
  "COMMON.LAST.3.MONTHS": "Son 3 Ay",
  "COMMON.LAST.6.MONTHS": "Son 6 Ay",
  "COMMON.LOGIN": "Giriş",
  "COMMON.LOGOUT": "Çıkış",
  "COMMON.OK": "Tamam",
  "COMMON.PLEASE.WAIT": "Lütfen Bekleyin",
  "COMMON.PROCESS": "İşlem",
  "COMMON.PTTBANK": "PTTBank",
  "COMMON.RECIPIENT.DETAILS": "Alıcı Bilgileri",
  "COMMON.SEARCH": "Ara",
  "COMMON.SELECT.DATE": "Tarih Seç",
  "COMMON.SUBMIT": "Gönder",
  "COMMON.THIS.MONTH": "Bu Ay",
  "COMMON.THIS.WEEK": "Bu Hafta",
  "COMMON.THIS.YEAR": "Bu Yıl",
  "COMMON.TODAY": "Bugün",
  "COMMON.TOMORROW": "Yarın",
  "COMMON.TRY_AGAIN": "Lütfen daha sorna tekrar deneyin!",
  "COMMON.UNKNOWN": "Bilinmiyor",
  "COMMON.WARNING": "Uyarı",
  "CVC.CODE": "CVC Kodu",
  "CVC.INFO.DESCRIPTION": "Kart Doğrulama Kodu veya CVC*, banka veya kredi kartınızın üzerine basılan ekstra bir koddur. Kartınızın arka yüzündeki imza şeridinde yazılı olan numaranın son üç hanesidir.",
  "CVC.INFO.TITLE": "CVC - Kart Doğrulama Kodu",
  "DAY.1": "Pazartesi",
  "DAY.2": "Salı",
  "DAY.3": "Çarşamba",
  "DAY.4": "Perşembe",
  "DAY.5": "Cuma",
  "EXPIRE.DATE.PLACEHOLDER": "Ay / Yıl",
  "EXPIRE.DATE.TITLE": "Son Kullanım Tarihi",
  "INPUT.ACCOUNTNUMBER": "Hesap Numarası",
  "INPUT.AMOUNT": "Gönderilecek Tutar",
  "INPUT.BANKACCOUNTNUMBER": "Banka Hesap Numarası",
  "INPUT.CARD.INFO": "Kart Bilgileri",
  "INPUT.CREDITCARD": "Kredi kartı",
  "INPUT.CUSTOMERNUMBER": "Müşteri Numarası",
  "INPUT.CVC": "CVC",
  "INPUT.DATE": "Tarih",
  "INPUT.DATETIME": "Tarih/Zaman",
  "INPUT.DROPDOWN": "Seçim Yap",
  "INPUT.DROPDOWN.SEARCH": "{name} Ara ...",
  "INPUT.EXPIREDATE": "Son kullanım tarihi",
  "INPUT.IBAN": "Iban",
  "INPUT.INSTITUTION.NAME": "Kurum Adı",
  "INPUT.MYACCOUNT": "Hesabım",
  "INPUT.NAME": "Ad Giriniz",
  "INPUT.NUMERIC": "Numara",
  "INPUT.PASSWORD": "Şifre",
  "INPUT.PASSWORD.CHANGE.PERIOD": "Şifre Yenileme Süresi",
  "INPUT.PASSWORD.CONFIRM": "Şifre Doğrulama",
  "INPUT.PHONE": "Telefon",
  "INPUT.PHONE.NUMBER": "Telefon Numarası",
  "INPUT.PINPASSWORD": "Pin şifresi",
  "INPUT.PLATENUMBER": "Plaka",
  "INPUT.RECORDNAME": "Hızlı İşlem Adı Giriniz",
  "INPUT.SELECT.INPUT": "{name} Seçiniz",
  "INPUT.STATEMENT": "Açıklama Giriniz",
  "INPUT.SUBSCRIBER.NUMBER": "Abone Numarası",
  "INPUT.SURNAME": "Soyad Giriniz",
  "INPUT.TCNUMBER": "TC Kimlik Numarası",
  "INPUT.TIME": "Zaman",
  "INPUT.USERNUMBER": "Kullanıcı Numarası",
  "LIST.EMPTY.WARN": "Liste boş",
  "LIST.ITEM.FAVORITE.ADD": "Favoriye Ekle",
  "LIST.ITEM.FAVORITE.REMOVE": "Favoriden Çıkart",
  "LOGOUT.DESCRIPTION": "Çıkış yapmak istediğinize emin misiniz?",
  "LOGOUT.TITLE": "Çıkış Yap",
  "NEW.PASSWORD.SCREEN.SUBMIT": "Yeni Şifreyi Kaydet",
  "NEW.PASSWORD.SCREEN.TITLE": "Yeni Şifre Oluştur",
  "NOTIFICATION.CLEAR.ALL": "Tümünü Sil",
  "NOTIFICATION.EMPTY.WARN": "Bugün veya öncesinde dönem içi bildiriminiz bulunmamaktadır!",
  "NOTIFICATION.TITLE": "Bildirimler",
  "OTP.DIGIT.REQUIREMENT": "Tam olarak 6 rakam olmalıdır",
  "OTP.INSTRUCTIONS": "Lütfen {phoneNumber} no'lu
telefonunuza gelen şifreyi giriniz.",
  "OTP.INSTRUCTIONS.NO.PHONE": "Kayıtlı telefon adresinize gelen SMS doğrulama kodunu giriniz.",
  "OTP.REMAININGTIME": "Doğrulama için kalan süre",
  "OTP.RESENDCODE": "Tekrar Gönder",
  "OTP.SMS.VERIFICATION.CODE": "SMS doğrulama kodu",
  "OTP.SMSNOTRECEIVED": "SMS ulaşmadı mı ?",
  "OTP.TITLE": "Sms Şifre Doğrulama",
  "OTP.VERIFY": "Doğrula",
  "PROFILE.ARENTYOU": "Siz Değil Misiniz?",
  "PROFILE.LOGOUT.DESCRIPTION": "Başka bir hesap ile giriş yapmak istediğinize emin misiniz?",
  "PROFILE.LOGOUT.TITLE": "Hesap Değişikliğini Onaylayın",
  "SELECT.MONTH": "{month} Ay",
  "TEST": "TEST",
  "TEST.WITH.PARAM": "TEST WITH {testText}",
  "VALIDATION.CONFIRMATION": "{name} doğrulaması gerekli.",
  "VALIDATION.CONFIRMATION_NOT_MATCH": "{name} ve {confirmName} alanları uyuşmuyor.",
  "VALIDATION.EMAIL_FORMAT": "Geçersiz eposta formatı.",
  "VALIDATION.INVALID": "Geçersiz {name}.",
  "VALIDATION.INVALID.AMOUNT_NOT_MULTIPLE_OF_10": "Tutar 10'un katları olmalıdır.",
  "VALIDATION.INVALID.FORMAT": "Geçersiz {name} formatı.",
  "VALIDATION.IP_ADDRESS": "Geçerli bir IP adresi girmediniz",
  "VALIDATION.MAX_LENGTH": "En fazla {length} karakter.",
  "VALIDATION.MAX_NUMBER": "{name} alanı {number} sayısından fazla olamaz.",
  "VALIDATION.MIN_LENGTH": "En az {length} karakter.",
  "VALIDATION.NUMBER": "{name} rakam olmalıdır.",
  "VALIDATION.POSITIVE": "{name} alanı pozitif sayı olmalıdır.",
  "VALIDATION.REQUIRED": "{name} alanı zorunludur.",
  "VALIDATION.SELECT": "Seçim yapmalısınız.",
  "VERIFY.CARD.INFORMATION.AGREEMENT": "{link} okudum, onaylıyorum",
  "VERIFY.CARD.INFORMATION.AGREEMENT.CONTENTS": "İnternet Bankacılığı İçerik",
  "VERIFY.CARD.INFORMATION.AGREEMENT.LINK": "İnternet Bankacılığı Sözleşmesini",
  "VERIFY.CARD.INFORMATION.AGREEMENT.TITLE": "İnternet Bankacılığı Sözleşmesi",
  "VERIFY.CARD.INFORMATION.PTT.CARD.NUMBER": "PTTKart Numarası",
  "VERIFY.CARD.INFORMATION.PTT.CARD.PASSWORD": "PTTKart Şifresi",
  "VERIFY.CARD.INFORMATION.REQUIRED.CARD.NUMBER": "Kart Numarası Zorunlu",
  "VERIFY.CARD.INFORMATION.REQUIRED.CVC": "CVC Kodu Zorunlu",
  "VERIFY.CARD.INFORMATION.REQUIRED.EXPIRE.DATE": "Son Kullanım Tarihi Zorunlu",
  "VERIFY.CARD.INFORMATION.REQUIRED.PIN.PASSWORD": "PTTKart Şifresi Zorunlu",
  "VERIFY.CARD.INFORMATION.TITLE": "Kart Bilgilerini Doğrula",
  "VERIFY.CARD.INFORMATION.VERIFY": "Doğrula",
}
`;
